core/src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -145,6 +145,17 @@ public Object eval(DataInstance model, EvaluationContext evalContext) {
 
         Hashtable funcHandlers = evalContext.getFunctionHandlers();
 
+
+        //check for custom handler, use this if it exists.
+        try {
+            IFunctionHandler handler = (IFunctionHandler) funcHandlers.get(name);
+            if (handler != null) {
+                return evalCustomFunction(handler, argVals, evalContext);
+            }
+        } catch(XPathArityException e){
+            // we matched the name but not the arg count.
+            // continue in case the default has the right arity.
+        }

```

Comment:
in https://github.com/dimagi/touchforms/pull/89/files#diff-752ce0230860c47c4ce3a737af9ce26fR54 we define the return type as broadly as possible (class) so that we can return different types. will that work with this or do we need to be more specific in our overrides?